eof/dataspace_client.py

```diff
@@ -287,18 +291,12 @@ def query_orbit_file_service(query: str) -> list[dict]:
     return query_results
 
 
-def get_access_token(username, password, token_2fa, netrc_file) -> Optional[str]:
+def get_access_token(username, password, token_2fa) -> str:
     """Get an access token for the Copernicus Data Space Ecosystem (CDSE) API.
 
     Code from https://documentation.dataspace.copernicus.eu/APIs/Token.html
     """
-    if not (username and password):
-        logger.debug("Get credentials form netrc")
-        try:
-            username, password = get_netrc_credentials(DATASPACE_HOST, netrc_file)
-        except FileNotFoundError:
-            logger.warning("No netrc file found.")
-            return None
+    assert username and password, "Username and password values are expected!"
```

Comment:
```suggestion
    if not (username and password):
        raise ValueError("Username and password values are expected!")
```

Reply:
I've considered `username` and `password`  to be set as part of the function contract: if this precondition is not valid, then there is a programming error, and it's up to us to prevent this situation from happening. 

Given `username` and `password` being set are a postcondition of `get_netrc_credentials()` (the function cannot return none values), which is called just before, everything is fine so far. Perfect DbC in action -- well... almost if we ignore I forgot to document the contracts in the docstring of each function, and that I should have tested for them to not be empty as well. ^^'

I usually dislike wide contracts that check every single things that are not meant to happen and throw logic errors if they do. That has a tendency to complexify source code: we add a lot of dead code that can/should never execute.

Here is the fixed code and docstring. If you really prefer wide contract instead of narrow ones. I'll throw then -- an `AssertionError` would still make more sense IMO.

Reply:
My mistake. The 3 scenarios weren't correctly handled. This should be better now.